pyshacl/constraints/core/shape_based_constraints.py

```diff
@@ -110,9 +110,15 @@ def _evaluate_property_shape(
         if potentially_recursive and prop_shape in potentially_recursive:
             warn(ShapeRecursionWarning(_evaluation_path))
             return _non_conformant, _reports
-        if not prop_shape or not prop_shape.is_property_shape:
+        if not prop_shape:
             raise ReportableRuntimeError(
-                "Shape pointed to by sh:property does not exist or is not a well-formed SHACL PropertyShape."
+                f"SHACL PropertyShape not found: The shape referenced by sh:property does not exist. "
+                f"Please check if the shape '{prop_shape}' is defined."
+            )
+        elif not prop_shape.is_property_shape:
+            raise ReportableRuntimeError(
+                f"Invalid SHACL PropertyShape: The shape '{prop_shape}' exists but is not a valid SHACL PropertyShape. "
```

Comment:
Can you please use the terminology "not a well-formed SHACL PropertyShape", that is how it is written in the SHACL spec.

Reply:
Sure thing, updated the error message